frontend/src/pages/signUp/SignUp.tsx

```diff
@@ -0,0 +1,160 @@
+                  style={{
+                    borderRadius: '8px',
+                    border: '1px solid #D8E0E8',
+                    background: '#F8F9FD',
+                    marginBottom: '1rem'
+                  }}
+                  placeholder="First Name"
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                    updateData(e);
+                  }}
+                  required
+                />
+              </div>
+              <div>
+                <InputLabel htmlFor="last-name" style={{ color: '#55555F' }}>
+                  Last Name
+                </InputLabel>
+                <TextField
+                  id="last-name"
+                  name="lastName"
+                  value={data.lastName}
+                  style={{
+                    borderRadius: '8px',
+                    border: '1px solid #D8E0E8',
+                    background: '#F8F9FD',
+                    marginBottom: '1rem'
+                  }}
+                  placeholder="Last name"
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                    updateData(e);
+                  }}
+                  required
+                />
+              </div>
+            </div>
+            <InputLabel htmlFor="email" style={{ color: '#55555F' }}>
+              Email
+            </InputLabel>
+            <TextField
+              id="email"
+              name="email"
+              value={data.email}
+              style={{
+                borderRadius: '8px',
+                border: '1px solid #D8E0E8',
+                background: '#F8F9FD',
+                marginBottom: '1rem'
+              }}
+              placeholder="Email"
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                updateData(e);
+              }}
+              required
+            />
+            <InputLabel htmlFor="password" style={{ color: '#55555F' }}>
+              Password
+            </InputLabel>
+            <TextField
+              id="password"
+              name="password"
+              value={data.password}
+              style={{
+                borderRadius: '8px',
+                border: '1px solid #D8E0E8',
+                background: '#F8F9FD',
+                marginBottom: '1rem'
+              }}
+              placeholder="Password"
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                updateData(e);
+              }}
+              type="password"
+              required
+            />
+            <div>
+              <Checkbox /> Remeber me
+            </div>
+            <Button
+              type="submit"
+              variant="contained"
+              style={{
+                color: '#F1F2F4',
+                borderRadius: '8px',
+                height: '3rem',
+                marginBottom: '1rem'
+              }}
+              disabled={!Object.values(data).every(Boolean)}>
```

Comment:
Isn't there another possibility to check if the form is valid? https://react-hook-form.com/get-started#Applyvalidation

Reply:
Did it.